Startup.cs

```diff
@@ -1,59 +1,91 @@
-		{
-			loggerFactory.AddConsole(minLevel: LogLevel.Warning);
-
-			// custom middleware to re-route all paths starting with `/app` to `/index.html`
-			applicationBuilder.Use(async (context, next) =>
-			{
-				if (context.Request.Path.StartsWithSegments(new PathString("/app")))
-					context.Request.Path = new PathString("/index.html");
-				await next();
-			});
-			applicationBuilder.UseApplicationInsightsRequestTelemetry();
-			applicationBuilder.UseApplicationInsightsExceptionTelemetry();
-			applicationBuilder.UseDefaultFiles();
-			applicationBuilder.UseStaticFiles();
-		}
-	}
-}
+using System;
+using System.IO;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Logging;
+using Microsoft.AspNetCore.Mvc.Razor;
+using Microsoft.CodeAnalysis;
+
+namespace Zoltu.Bags.Client
+{
+	public class Startup
+	{
+		public static void Main(String[] args)
+		{
+			new WebHostBuilder()
+				.UseKestrel()
+				.UseContentRoot(Directory.GetCurrentDirectory())
+				.UseWebRoot("client")
+				.UseStartup<Startup>()
+				.UseUrls("http://*:80")
+				.Build()
+				.Run();
+		}
+
+		private IConfiguration _configuration;
+
+		public Startup(IHostingEnvironment hostingEnvironment)
+		{
+			_configuration = new ConfigurationBuilder()
+				.AddApplicationInsightsSettings(developerMode: hostingEnvironment.IsDevelopment())
+                       .SetBasePath(hostingEnvironment.ContentRootPath)
+                .AddJsonFile("appsettings.json", optional: true, reloadOnChange: true)
+                .AddJsonFile($"appsettings.{hostingEnvironment.EnvironmentName}.json", optional: true)
+                .AddEnvironmentVariables()
+				.Build();
+		}
+
+		public void ConfigureServices(IServiceCollection services)
+		{
+			services.AddApplicationInsightsTelemetry(_configuration);
+            services.AddMvc();
+
+            services.Configure<RazorViewEngineOptions>(options =>
+            {
+                var previous = options.CompilationCallback;
+                options.CompilationCallback = (context) =>
+                {
+                    previous?.Invoke(context);
+                };
+            });
+        }
+
+		public void Configure(IApplicationBuilder applicationBuilder, IHostingEnvironment hostingEnvironment, ILoggerFactory loggerFactory)
+		{
+			loggerFactory.AddConsole(minLevel: LogLevel.Warning);
+
+            // custom middleware to re-route all paths starting with `/app` to `/index.html`
+            //applicationBuilder.Use(async (context, next) =>
+            //{
+            //	if (context.Request.Path.StartsWithSegments(new PathString("/app")))
+            //		context.Request.Path = new PathString("/index.html");
+            //	await next();
+            //});

```

Comment:
Delete this code if it is unused.  However, I suspect this code (or something similar is actually necessary to get all of the routes hooked up.  At the moment it appears that the hooked up route is ` [Route("app", Name = "app", Order = 0)]`, which I'm not sure will catch everything inside of `/app`.  We need the controller action to be called for every route that starts with `/app`, which includes `/app/aboutus`, `/app/foo`, etc.  We may break some of those out into separate manual routes later but for now we want a single catch-all route that handles everything starting with `/app`.

One way to achieve this may be to leave this middleware in place (uncomment it) and make `/index.html` be the only route handled by the controller action.  The middleware will trigger first, redirecting everything in `/app` to `/index.html` and then MVC will pick up that route and render it.